docs/services/garage.md

```diff
@@ -0,0 +1,104 @@
+# Miniflux
+
+[Garage](https://garagehq.deuxfleurs.fr/) is an open-source distributed object storage service tailored for self-hosting.
+
+
+## Dependencies
+
+This service requires the following other services:
+
+- a [Traefik](traefik.md) reverse-proxy server
+
+
+## Configuration
+
+To enable this service, add the following configuration to your `vars.yml` file, adjust it to your needs and re-run the [installation](../installing.md) process:
+
+```yaml
+########################################################################
+#                                                                      #
+# garage                                                               #
+#                                                                      #
+########################################################################
+
+# Garage uses a secret key that is shared between all nodes of the cluster in order to identify these nodes and allow
+# them to communicate together. This key should be specified here in the form of a 32-byte hex-encoded random string.
+# Such a string can be generated with a command such as ´openssl rand -hex 32´.
+garage_rpc_secret: 'verysecret'
+
+# Token for accessing the admin API.
+# Generating a strong secret is advised (e.g. `openssl rand -hex 32`).
+garage_admin_token: 'verysecret'
+
+# This is the zone that your host will be in. Refer to https://garagehq.deuxfleurs.fr/documentation/cookbook/real-world/#creating-a-cluster-layout
+# for more information
+garage_zone: "dc1"
+
+# The replication mode defines how many copies of your data are stored per region
+# "2" means data stored on Garage will be stored on two different nodes, if possible in different zones. Adjust this
+# to your needs.
+# Full reference: https://garagehq.deuxfleurs.fr/documentation/reference-manual/configuration/#replication-mode
+garage_replication_mode: 2
+
+
+########################
+## NODE CONFIGURATION ##
+########################
+
+# This is an example configuration of nodes. On this host we run 2 data nodes that both manage 1 3TB drive.
+# We access the nodes via a gateway node.
+
+garage_node1_base_path: "/media/drive1/garage/node1"
+garage_node2_base_path: "/media/drive2/garage/node2"
+garage_nodes:
+  - name: "gateway1"
+    metadata_path: "{{ garage_meta_path }}/gw1"
+    data_path: "{{ garage_data_path }}/gw1"
+    gateway: true
+    rpc_bind_port: 3901
+    node_addr: "garage-gw1.example.com"
+    s3_api_addr: "s3.example.com"
+  - name: "node1"
+    gateway: false
+    capacity: 3
+    metadata_path: "{{ garage_node1_base_path }}/metadata"
+    data_path: "{{ garage_node1_base_path }}/data"
+    rpc_bind_port: 3911
+    node_addr: "garage-node1.example.com"
+  - name: "node2"
+    gateway: false
+    capacity: 3
+    metadata_path: "{{ garage_node2_base_path }}/metadata"
+    data_path: "{{ garage_node2_base_path }}/data"
+    rpc_bind_port: 3921
+    node_addr: "garage-node2.example.com"
```

Comment:
I suppose this configuration is somewhat confusion, because it's a bit of a contrived example?

Is there a use-case where one would want to run both the gateway and 2 nodes on the same server?

I suppose it would be nice to see an example of how to host each node separately and then add them all together. Perhaps a layout like this would be nice:

- one server which is both a gateway and a node (or 2 nodes, for making it even more complicated)
- one other server which is just a node

And then seeing how these 2 separate servers are wired together and integrated into the gateway.

Do each node need to know about each other node? Or can each node be completely unaware of the others (configuration-wise) and just talk to the gateway to discover the other nodes?

Reply:
This is more or less my actual setup. The idea is that for multiple drives on one server you run multiple garage daemons, this is discussed here: https://git.deuxfleurs.fr/Deuxfleurs/garage/issues/218

But I think it would be nice to show in this documentation two separate servers that are wired together. Will do.

The nodes can connect by talking to the gateway (or any other node) as long as they share the RPC secret